test/test_helper_discovery.rb

```diff
@@ -0,0 +1,25 @@
+def set_session_user_with_perms perms
+  perms = perms.collect{|p| Permission.find_by_name(p) || Permission.create(:name => p) }
+  perms.each do |p|
+    p.resource_type = 'Host' if p.name =~ /discovered_hosts$/
+    p.resource_type = 'DiscoveryRule' if p.name =~ /discovery_rules$/
+    p.save!
+  end
+  role = FactoryGirl.create :role
+  perms.each do |perm|
+    filter = FactoryGirl.create(:filter, :role => role, :permissions => [perm])
+    filter.save!
```

Comment:
I understand this is optimized but still if you aren't doing anything I don't see why it's here, unless this is in case the create somehow fails and then the save! will probably fail as well.